druid/src/widget/split.rs

```diff
@@ -0,0 +1,311 @@
+                    ),
+                )
+            }
+        };
+        let child1_size = self.child1.layout(ctx, &child1_bc, &data, env);
+        let child2_size = self.child2.layout(ctx, &child2_bc, &data, env);
+
+        //Top-left align for both children, out of laziness.
+        //Reduce our unsplit direction to the larger of the two widgets
+        let (child1_rect, child2_rect) = match self.split_direction {
+            Axis::Horizontal => {
+                my_size.height = child1_size.height.max(child2_size.height);
+                (
+                    Rect::from_origin_size(Point::ORIGIN, child1_size),
+                    Rect::from_origin_size(
+                        Point::new(child1_size.width + self.splitter_size, 0.0),
+                        child2_size,
+                    ),
+                )
+            }
+            Axis::Vertical => {
+                my_size.width = child1_size.width.max(child2_size.width);
+                (
+                    Rect::from_origin_size(Point::ORIGIN, child1_size),
+                    Rect::from_origin_size(
+                        Point::new(0.0, child1_size.height + self.splitter_size),
+                        child2_size,
+                    ),
+                )
+            }
+        };
+        self.child1.set_layout_rect(child1_rect);
+        self.child2.set_layout_rect(child2_rect);
+        my_size
+    }
+
+    fn event(&mut self, event: &Event, ctx: &mut EventCtx, data: &mut T, env: &Env) {
+        if self.child1.state.is_active() {
+            self.child1.event(event, ctx, data, env);
+            if ctx.is_handled() {
+                return;
+            }
+        }
+        if self.child2.state.is_active() {
+            self.child2.event(event, ctx, data, env);
+            if ctx.is_handled() {
+                return;
+            }
+        }
+        if self.draggable {
+            match event {
+                Event::MouseDown(mouse) => {
+                    if mouse.button.is_left()
+                        && self.splitter_hit_test(ctx.base_state.size(), mouse.pos)
+                    {
+                        ctx.set_active(true);
+                        ctx.set_handled();
+                    }
+                }
+                Event::MouseUp(mouse) => {
+                    if mouse.button.is_left() && ctx.is_active() {
+                        ctx.set_active(false);
+                        self.update_splitter(ctx.base_state.size(), mouse.pos);
+                        ctx.invalidate();
+                    }
+                }
+                Event::MouseMoved(mouse) => {
+                    if ctx.is_active() {
+                        if !ctx.is_hot() {
+                            ctx.set_active(false);
+                        }
+                        self.update_splitter(ctx.base_state.size(), mouse.pos);
+                        ctx.invalidate();
+                    } else if ctx.is_hot()
```

Comment:
I think we should be setting the resize cursor while a drag is active, as well as when inactive but where we are in the hit-area of a split.